portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -69,10 +72,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
         if(queryVideoEntity == null){
             Video video;
             try {
-                video = Video.httpRequestToVideo(request);
-            } catch (MalformedURLException e) {
-                sendErrorMessage(response, HttpServletResponse.SC_NOT_FOUND, "Invalid thumbnail URL.");
+                video = AUX.VideoIdToObject(id);
+            } catch (IOException e) {
+                sendErrorMessage(response, HttpServletResponse.SC_NOT_FOUND, "Unable to retrieve video.");
+                return;
+            } catch (IndexOutOfBoundsException e) {
```

Comment:
Index out of bounds currently handled here.

Reply:
Maybe this will all go away once we have a unified VideoException?

Reply:
Yep, will check back on that when we have one.